classes/db/DAO.php

```diff
@@ -330,10 +330,10 @@ public function convertFromDB($value, $type, $nullable = false)
                 return (float) $value;
             case 'object':
             case 'array':
-                $decodedValue = json_decode($value, true);
+                $decodedValue = json_decode($value ?? ($type === 'array' ? '[]' : '{}'), true);
                 // FIXME: pkp/pkp-lib#6250 Remove after 3.3.x upgrade code is removed (see also pkp/pkp-lib#5772)
                 if (!is_null($decodedValue)) {
-                    return $decodedValue;
+                    return $type === 'array' ? array_values((array) $decodedValue) : (object) $decodedValue;
```

Comment:
For here and below, the database might be holding a valid JSON, with an unexpected value.
For example, if we encode the array `[1 => 0]` using the `$type` `array`, it will be stored as a JSON object.

There are already some known cases out in the wild.